alexandria/core/tests/test_views.py

```diff
@@ -398,6 +399,70 @@ def test_move_document_to_new_category(
     assert response.status_code == HTTP_200_OK
 
 
+@pytest.mark.parametrize(
+    "to_category,expected_status",
+    [
+        ("same", HTTP_201_CREATED),
+        ("new", HTTP_201_CREATED),
+        ("not_defined", HTTP_201_CREATED),
+        ("non_existent", HTTP_400_BAD_REQUEST),
+        ("not_allowed", HTTP_400_BAD_REQUEST),
+    ],
+)
+def test_copy_document(
+    admin_client,
+    category_factory,
+    file_factory,
+    document_factory,
+    to_category,
+    expected_status,
+):
+    category_not_allowed = category_factory.create(allowed_mime_types=["plain/text"])
+    category = category_factory()
+    document = document_factory(category=category)
+    file_factory.create(document=document, name="Image.jpeg", mime_type="image/jpeg")
+
+    if to_category == "non_existent":
+        target_category_pk = 999
+    elif to_category == "new":
+        target_category_pk = category_factory().pk
+    elif to_category == "not_allowed":
+        target_category_pk = category_not_allowed.pk
+    else:
+        target_category_pk = document.category.pk
+
+    request_body = {
+        "data": {
+            "category": str(target_category_pk),
+            "type": "documents",
+            "id": str(document.pk),
+        },
+    }
+
+    if to_category == "not_defined":
+        request_body["data"].pop("category")
+
+    url = reverse("document-copy", args=[document.pk])
+    response = admin_client.post(
+        url,
+        data=json.dumps(request_body),
```

Comment:
That shouldn't be necessary.

`admin_client.post(url, data=request_data)` should just work

Reply:
Changed now